ykrt/src/compile/jitc_yk/codegen/x64/mod.rs

```diff
@@ -347,14 +348,45 @@ impl<'a> Assemble<'a> {
                 }
                 dynasm!(self.asm; mov rcx, rsp);
 
+                // Push [GuardIdx]'s of previous guard failures.
+                if let Some(ref guards) = prevguards {
```

Comment:
I might have misunderstood what's going on here, but isn't this producing at run-time an ever-growing list of guard IDs? Looked at another way, does this mean we get a gradual slow-down at run-time as you go deeper into the side-trace tree?

Reply:
Sort-of yes. But this only happens during guard failures, so I'm not sure how much a few extra pushes hurt here. How deeply nested to we expect side-traces to become?

Reply:
We can deal with it for now but this will become a signifcant overhead at some point, even for a single level of nesting, and we can regularly expect to have 3-5 levels deep at least I would expect.

What we'll need to do -- not in this PR! -- is work out how to make this a JIT-compile-time cost that has zero impact on the running trace.